### PR TITLE
[JENKINS-42950] Making users more aware of behavior

### PR DESCRIPTION
**Pull request by jglick**
[JENKINS-42950](https://issues.jenkins-ci.org/browse/JENKINS-42950)

### CODE REVIEW

File: src/main/resources/org/jenkinsci/plugins/credentialsbinding/impl/BindingStep/help.html [hunk 1]
```diff
@@ -58,6 +58,29 @@
...
++ echo 'foo'\''bar'
+foo'bar
+</code></pre>
+<p>
+    Mangled secrets are not detected by Jenkins and will appear in plain text in the log file, in this case as:
+</p>
+<pre><code>+ echo 'foo'\''bar'
+****</code></pre>
+<p>
+    This particular issue can be worked around by turning off echo with <code>set +x</code>,
```
Comment (jglick) [inline src/main/resources/org/jenkinsci/plugins/credentialsbinding/impl/BindingStep/help.html]: Note that [`sh` sets `-xe`](https://github.com/jenkinsci/durable-task-plugin/blob/c8e1d57986ecc857a0b74f4b21da2b64dc27e3f4/src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java#L118), following the example of [freestyle builds](https://github.com/jenkinsci/jenkins/blob/183ca1f95ebc44318194ff53887b876e47d92d34/core/src/main/java/hudson/tasks/Shell.java#L97), which is undesirable when it comes to secrets.

File: src/main/resources/org/jenkinsci/plugins/credentialsbinding/impl/BindingStep/help.html [hunk 2]
```diff
@@ -58,6 +58,29 @@
     anyone permitted to configure a job or define Pipeline steps
     is assumed to be trusted to use any credentials in scope however they like.
 </p>
+<p>
+    Beware that certain tools mangle secrets when displaying them.
+    For example, Bash (as opposed to Ubuntu’s plainer Dash) does so with text containing <code>'</code> in echo mode:
```
[the remark this answers is not in the document]
Comment (daniel-beck) [reply src/main/resources/org/jenkinsci/plugins/credentialsbinding/impl/BindingStep/help.html]: This still seems misleading, even if perhaps only to a non-native speaker?

`s/ in/ or similar metacharacters in` ?

**Comment by jglick** [comment]
Also see: https://github.com/jenkins-infra/jenkins.io/pull/2118

File: src/main/resources/org/jenkinsci/plugins/credentialsbinding/impl/BindingStep/help.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (jglick) [reply src/main/resources/org/jenkinsci/plugins/credentialsbinding/impl/BindingStep/help.html]: `$` does not seem to cause a problem here. Newlines do result in Bash quoting the argument (without escaping the newline), meaning that the secret will be literally present in the output yet not escaped by the line-oriented masker; this seems like a situation users are unlikely to encounter. I think it suffices to discuss `'`.

**Pull request merged**